MAINTAINERS.md

```diff
@@ -19,3 +19,65 @@ This document lists the maintainers of the Project. Maintainers may be added onc
 | Clearvive     | Clearvive           | Datastrato       |
 | Cheyne        | ch3yne              | Datastrato       |
 | Jerry Shao    | jerryshao           | Datastrato       |
+
+## Review process
+
+All contributions to the project must be reviewed by **AT LEAST** one maintainer before merging.
+For larger changes, it is recommended to have more than one reviewer. In particular, if you are
+working on a area you are not familiar with, it is recommended to ask for review from that area by
+using `git log --format=full <filename>` to see who committed the most.
+
+## When to commit/merge a pull request
+
+PRs shall not be merged during active, ongoing discussions, unless they address issues such as
+critical security fixes or public vulnerabilities. Time should be given prior to merging for
+those involved with conversations to be resolved.
+
+## How to merge a pull request
+
+Changes pushed to the main branch on Gravitino cannot be **removed**, that is, we can't do
+force-push to it. So please don't add any test commits or anything like that. If the PR is
+merged by mistake, please using `git revert` to revert the commit, not `git reset` or anything
+like this.
+
+Please use the "Squash and merge" option when merging a PR. This will keep the commit history
+clean and meaningful.
+
+Gravitino use "issue" to track the project progress. So make sure each PR has a related issue, if
+not, please ask the PR author to create one. Unless this PR is a trivial one, like fixing a typo or
+something like that, all PRs should have related issues.
+
+1. Before merging, make sure the PR is approved by **at least** one maintainer.
```

Comment:
Maybe we should check the milestone of pull request and issue before we merge the pull request.

Reply:
No, PR doesn't have a milestone. In 0.5.0 we use the milestone to manage issues, but due to the limitation of Github that one issue can only have one milestone, so from 0.6.0, I plan to use labels to manage fixed versions, which can support multiple versions.

Reply:
OK.